File: 7thSeaEditor/7thSeaEditor/SeventhSeaPart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _7thSeaEditor
{
    class SeventhSeaPart
    {
        int id;
        int categoryId;
        string name;
        string description;

        public SeventhSeaPart(int id, string name, string description, int categoryId)
        {
            this.id = id;
            this.name = name;
            this.description = description;
            this.categoryId = categoryId;
        }

        public string GetMarkdown()
        {
            return Markdown(name, description);
        }

        public int Id
        {
            get { return id; }
        }

        public string Name
        {
            get { return name; }
            set { name = value; }
        }

        public string Description
        {
            get { return description; }
            set { description = value; }
        }

        public static string Markdown(string name, string description)
        {
            if (description == "")
                return "<h3>" + name + "</h3>";
            else if (name == "")
                return description;
            else
                return "<strong>" + name + ":</strong>" + description;
        }
    }
}

File: 7thSeaCompendium/src/com/seventhsea/compendium/SeventhSeaPart.java
package com.seventhsea.compendium;

public class SeventhSeaPart {
	private int id;
	private String name;
	private String description;
	
	public SeventhSeaPart(int id, String name, String description){
		this.id = id;
		this.name = name;
		this.description = description;
	}
	
	public int getId(){
		return id;
	}
	
	public String getName(){
		return name;
	}
	
	public String getDescription(){
		return description;
	}
	
	public String getMarkup(){
        if (description.length() == 0)
            return "<h3>" + name + "</h3>";
        else if (name.length() == 0)
            return description;
        else
            return "<strong>" + name + ":</strong>" + description;
	}
}
